src/test/ui/wf/wf-unsafe-trait-obj-match.stderr

```diff
@@ -11,6 +11,10 @@ LL | |     }
    |
    = note: expected reference `&S`
               found reference `&R`
+help: consider using a semicolon here, but this will discard any values in the match arms
+   |
+LL |     };
+   |      +
```

Comment:
Should the suggestion be restricted to the case where all the arms have compatible types or unit?

Reply:
We already have suggestions for those, but if you look at #72634 the issue comes up with a match statement with a single arm diverging to `!`. We could argue that we should mention "assign the match value to a let binding" or look at every match arm and suggest returning the value 🤔